plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py

```diff
@@ -57,7 +62,26 @@ def host_traps_string_to_list(self, hosts):
                     element["traps"] = list(element.get("traps").split())
             return hosts
 
-    def populate_facts(self, connection, ansible_facts, data=None):
+    def get_snmpv3_user_facts(self, snmpv3_user):
```

Comment:
As this, is not a trivial resource module way, can we add some docstrings and comment to the code for us to reference it better later

Reply:
Add a note to the user attribute in docs to state that the behavior is supposed to impact idempotency.